src/shared/cloudformation/activation.ts

```diff
@@ -58,12 +58,22 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
 }
 
 /**
- * Adds custom tags to the YAML extension's settings in order to hide error notifications for intrinsic functions if a user has the YAML extension.
- * Lifted near-verbatim from cfn-lint's VS Code extension; writes to workspace instead of global.
+ * Adds custom tags to the YAML extension's settings in order to hide error
+ * notifications for SAM/CFN intrinsic functions if a user has the YAML extension.
+ *
+ * Lifted near-verbatim from the cfn-lint VSCode extension.
  * https://github.com/aws-cloudformation/cfn-lint-visual-studio-code/blob/629de0bac4f36cfc6534e409a6f6766a2240992f/client/src/extension.ts#L56
  */
 function addCustomTags(): void {
-    const currentTags = vscode.workspace.getConfiguration().get<string[]>('yaml.customTags') ?? []
+    const settingName = 'yaml.customTags'
+    const currentTags = vscode.workspace.getConfiguration().get<string[]>(settingName) ?? []
+    if (!Array.isArray(currentTags)) {
+        getLogger().error(
+            'setting "%s" is not an array. SAM/CFN intrinsic functions will not be recognized.',
```

Comment:
Reminder that we can't trust the result of `getConfiguration().get()` (the type parameter is erased at runtime).

Reply:
Might be worth trying to handle this as an array? Something like https://github.com/aws/aws-toolkit-vscode/blob/7034750338fe7b94217390bf9276ed655dfe005f/src/shared/cloudformation/cloudformation.ts#L857

Reply:
> Might be worth trying to handle this as an array? Something like

But then we are potentially overwriting the user's settings with something unwanted.